Formula/o/opensca-cli.rb

```diff
@@ -0,0 +1,19 @@
+class OpenscaCli < Formula
+  desc "OpenSCA is a supply-chain security tool for security researchers and developers"
+  homepage "https://opensca.xmirror.cn"
+  url "https://github.com/XmirrorSecurity/OpenSCA-cli.git", tag: "v3.0.0", revision: "f5e43fcd2fdbf2de9c5326f01f0395738dfec1b8"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build",
+    *std_go_args(ldflags: "-s -w -X 'main.version=#{version}'"),
+    "-o", "#{bin}/opensca-cli", "main.go"
+  end
+
+  test do
+    output = shell_output("#{bin}/opensca-cli -version")
+    assert_match version.to_s, output
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.